src/transformers/models/blt/modular_blt.py

```diff
@@ -0,0 +1,1008 @@
+            attention_mask=causal_mask,
+            position_ids=position_ids,
+            past_key_values=None,
+            cache_position=None,
+            encoder_attention_mask=cross_attn_mask_enc,
+            num_patches=patch_lengths.shape[1],
+            patch_ids=patch_ids,
+            **kwargs,
+        )
+        global_hidden_states = encoder_cross_states.view(batch_size, patch_lengths.shape[1], -1)
+        global_cache_position = torch.arange(0, global_hidden_states.shape[1], device=global_hidden_states.device)
+        global_position_ids = global_cache_position.unsqueeze(0)
+        global_causal_mask = create_causal_mask(
+            config=self.config,
+            input_embeds=global_hidden_states,
+            attention_mask=None,
+            cache_position=global_cache_position,
+            past_key_values=None,
+            position_ids=None,
+        )
+
+        global_hidden_states = self.global_transformer(
+            input_embeds=global_hidden_states,
+            attention_mask=global_causal_mask,
+            position_ids=global_position_ids,
+            past_key_values=None,
+            cache_position=None,
+            **kwargs,
+        )
+        decoder_patch_ids = self._patch_ids_from_lengths(patch_lengths[:, 1:], sequence_length)
+        cross_attn_mask_dec = _prepare_patch_cross_attention_mask(
+            decoder_patch_ids,
+            patch_lengths.shape[1],
+            sequence_length,
+            False,
+            self.config.cross_attn_k,
+            encoder_embeds.dtype,
+        )
+        output = self.local_decoder(
+            input_ids=input_ids,
+            inputs_embeds=encoder_hidden_states,
+            patch_embeds=global_hidden_states,
+            attention_mask=causal_mask,
+            position_ids=position_ids,
+            past_key_values=past_key_values,
+            cache_position=cache_position,
+            encoder_attention_mask=cross_attn_mask_dec,
+            **kwargs,
+        )
+        return BaseModelOutputWithPast(
+            last_hidden_state=output,
+            past_key_values=past_key_values,
+        )
+
+    def get_input_embeddings(self):
+        return self.local_encoder.embed_tokens
+
+    def set_input_embeddings(self, value):
+        self.local_encoder.embed_tokens = value
+
+    def _patch_ids_from_lengths(self, patch_lengths: torch.Tensor, seq_len: int) -> torch.Tensor:
+        batch_size = patch_lengths.shape[0]
+        patch_starts = torch.cat(
+            [
+                torch.zeros(batch_size, 1, dtype=patch_lengths.dtype, device=patch_lengths.device),
+                patch_lengths.cumsum(dim=-1)[:, :-1],
+            ],
+            dim=-1,
+        )
+        token_positions = torch.arange(seq_len, device=patch_lengths.device)
+        return (patch_starts.unsqueeze(1) <= token_positions.unsqueeze(0).unsqueeze(-1)).sum(dim=-1) - 1
+
+
+class BltPatcher(BltPreTrainedModel):
+    def __init__(self, config: BltPatcherConfig):
```

Comment:
Missing the config annotation!
```suggestion

class BltPatcher(BltPreTrainedModel):
    config: BltPatcherConfig
    def __init__(self, config: BltPatcherConfig):
```